### ISSUE TITLE
python match case not highlighted

### ISSUE THREAD

**Issue by A-Walrus** [typst/typst#3604]
### Description

Python `match` and `case` keywords aren't highlighted.

![image](https://github.com/typst/typst/assets/58790821/92778039-f8d9-4921-a24d-52f474edae3f)

### Reproduction URL

https://typst.app/project/rvUJmuGPYUTVAyQavzkUEb

### Operating system

Web app, Linux

### Typst version

- [X] I am using the latest version of Typst

**Comment by A-Walrus** [comment]
Looking into it I understand this is probably upstream, but not sure where. I understand we take the syntaxes from bat, and they in turn take some of them from sublime?

Sublime text (looks correct):
![image](https://github.com/typst/typst/assets/58790821/35e91138-c6f9-49bb-aae9-463c3ec21bbe)

Bat (incorrect):
![image](https://github.com/typst/typst/assets/58790821/45642049-1dd7-4ad6-a629-07ec731c64f4)

**Comment by LaurenzV** [comment]
Yeah, as far as I know some of the Sublime Syntaxes are pretty outdated...

**Comment by freundTech** [comment]
Typst uses [syntect](https://github.com/trishume/syntect) for syntax highlighting. Typst also uses [two-face]https://github.com/cosmichorrordev/two-face) to provide additional syntaxes for syntect.
https://github.com/typst/typst/blob/0fef85706451331b00d09dc85a77a812c779ce6c/crates/typst/src/text/raw.rs#L825-L826
https://github.com/typst/typst/blob/0fef85706451331b00d09dc85a77a812c779ce6c/Cargo.toml#L109

two-face gets those syntaxes from [bat](https://github.com/sharkdp/bat) , which maintains the list for use in their own program. two-face includes those syntaxes using a git submodule pointing to a 9 months old version of bat (https://github.com/cosmichorrordev/two-face).

bat in turn gets its syntaxes from [Sublime Packages](https://github.com/sublimehq/Packages) and other sources. It uses a git submodule to include a 4 years old version of Sublime Packages.

**Comment by freundTech** [comment]
Ref: https://github.com/sharkdp/bat/issues/2895
